pkg/common/peertracker/listener_test.go

```diff
@@ -0,0 +1,98 @@
+package peertracker
+
+import (
+	"errors"
+	"io/ioutil"
+	"net"
+	"os"
+	"path"
+	"testing"
+
+	"github.com/stretchr/testify/suite"
+)
+
+var errMockWatcherFailed = errors.New("create new watcher failed")
+
+type failingMockTracker struct{}
+
+func (failingMockTracker) Close() {}
+func (failingMockTracker) NewWatcher(CallerInfo) (Watcher, error) {
+	return nil, errMockWatcherFailed
+}
+
+func newFailingMockTracker() (PeerTracker, error) {
+	return failingMockTracker{}, nil
+}
+
+func TestListenerTestSuite(t *testing.T) {
+	suite.Run(t, new(ListenerTestSuite))
+}
+
+type ListenerTestSuite struct {
+	suite.Suite
+
+	tempDir  string
+	ul       *Listener
+	unixAddr *net.UnixAddr
+}
+
+func (p *ListenerTestSuite) SetupTest() {
+	var err error
+
+	p.tempDir, err = ioutil.TempDir("", "spire-listener-test")
+	p.NoError(err)
```

Comment:
nit: p.Require().NoError since the test should not continue if we can't create the temp dir